tests/server/test_search_engine.py

```diff
@@ -136,3 +189,47 @@ async def test_create_index_with_existing_index(
 
         with pytest.raises(RequestError, match="resource_already_exists_exception"):
             await search_engine.create_index(dataset)
+
+    @pytest.mark.parametrize(
+        ("query", "expected_items"),
+        [
+            ("card", 2),
+            ("account", 1),
+            ("payment", 2),
+            ("cash", 2),
+            ("negative", 2),
+            ("00000", 1),
+            ("card payment", 2),
+            ("nothing", 0),
+            (SearchQuery(text=TextFieldFactory(q="card")), 2),
+            (SearchQuery(text=TextFieldFactory(q="account")), 1),
+            (SearchQuery(text=TextFieldFactory(q="payment")), 2),
+            (SearchQuery(text=TextFieldFactory(q="cash")), 2),
+            (SearchQuery(text=TextFieldFactory(q="card payment")), 2),
+            (SearchQuery(text=TextFieldFactory(q="nothing")), 0),
+            (SearchQuery(text=TextFieldFactory(q="negative", field="label")), 2),
+            (SearchQuery(text=TextFieldFactory(q="00000", field="textId")), 1),
+            (SearchQuery(text=TextFieldFactory(q="card payment", field="text")), 2),
```

Comment:
Okay, the test is failing because `TextFieldFactory` is being used outside the unit test function and the `db` fixture has not been executed yet and therefore the `scoped_session` does not have a `bind` assigned yet.

Instead of parametrizing the unit test with `SearchQuery(text=TextFieldFactory(...)`, we can pass the a dict with the parameters needs for creating the text field and then inside the unit test function, create the field using `TextFieldFactory`.